src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -165,24 +164,24 @@ public void setup() throws Exception {
 
   @After
   public void teardown() throws Exception {
+    logger.info(
+        "Teardown: trying to delete snapshots {}, dataset {}, billing profile {}",
+        snapshotIds,
+        datasetId,
+        profileId);
+
     dataRepoFixtures.resetConfig(steward);
-    snapshotIds.forEach(
-        snapshotId -> {
-          try {
-            dataRepoFixtures.deleteSnapshot(steward, snapshotId);
-            snapshotIds.remove(snapshotId);
```

Comment:
No need to clear out the contents of the snapshot ID list, or null out other resource IDs: besides being the source of our unexpected test behavior, subsequent test runs in the same suite will get a clean slate when the setup method runs.